packages/contracts/src/healthcheck/LossRatioHealthCheck.sol

```diff
@@ -0,0 +1,96 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity ^0.8.19;
+
+import {OwnableUpgradeable} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+
+import {SafeInitializable} from "../upgradeable/SafeInitializable.sol";
+import {IVersionable} from "../upgradeable/IVersionable.sol";
+import {IHealthCheck} from "./IHealthCheck.sol";
+import {SafeUUPSUpgradeable} from "../upgradeable/SafeUUPSUpgradeable.sol";
+
+    error HealthCheckFailed();
+
+contract LossRatioHealthCheck is SafeUUPSUpgradeable, IHealthCheck {
+    event ShutdownLossRatioChanged(uint256 ratio);
+    event HealthCheckEnabledChanged(bool enabled);
+
+    // represents 100%
+    uint256 public constant MAX_BPS = 10_000;
+
+    uint8 public constant PASS = 0;
+    uint8 public constant ACCEPTABLE_LOSS = 1;
+    uint8 public constant SIGNIFICANT_LOSS = 2;
+
+    // The ratio of the loss that will used to stop strategy.
+    uint256 public shutdownLossRatio;
+
+    /**
+ * @dev This empty reserved space is put in place to allow future versions to add new
+     * variables without shifting down storage in the inheritance chain.
+     * See https://docs.openzeppelin.com/contracts/4.x/upgradeable#storage_gaps
+     */
+    uint256[50] private __gap;
+
+    // ------------------------------------------ Constructors ------------------------------------------
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor(bool needDisableInitializers) SafeInitializable(needDisableInitializers) {} // solhint-disable-line no-empty-blocks
+
+    function initialize(uint256 _shutdownLossRatio)
+        public
+        initializer
+    {
+        __Ownable_init();
```

Comment:
You need init `SafeUUPSUpgradeable` there, ownable will be inited under the hood.
```suggestion
        __SafeUUPSUpgradeable_init();
```